File: Semana 4/Taller Modulos/tienda/usuario.js
class Usuario {
    constructor(nombre, email, rol = 'cliente', activo = true){
        this.nombre = nombre;
        this.email = email;
        this.rol = rol;
        this.activo = activo;
    }

    saludo() {
        return `Mi nombre es ${this.nombre} y mi rol ${this.rol}.`
    }

    desactivar() {
        return `El usuario fue desactivado. Activo: ${this.activo = false}`
    }

    resumen() {
        return `
                Nombre: ${this.nombre}
                Email: ${this.email}
                Rol: ${this.rol}
                Activo: ${this.activo}`
    }
}

class UsuarioVIP extends Usuario {
    constructor(nombre, email, membresia, activo = true){
        super(nombre, email, 'Cliente VIP', activo);
        this.membresia = membresia;
    }

    saludo() {
        return `${super.saludo()} Tengo la membresía ${this.membresia}`
    }

    beneficios() {
        return `Este nivel tiene los beneficios...`
    }
}

export { Usuario, UsuarioVIP };

File: Semana 8/OOP/src/Usuario.java
public class Usuario {
    private String email;
    private String role;

    // Constructor 1
    public Usuario (String email, String role) {
        this.email = email;
        this.role = role;
    }

    // Constructor 2
    public Usuario(String email) {
        this.email = email;
        this.role = "Invitado";
    }

    public String getEmail() {
        return email;
    }

    public String getRole() {
        return role;
    }
}

